docs/tutorial/writing-tests/test-case.rst

```diff
@@ -0,0 +1,1105 @@
+    ID: com.canonical.certification::network_iface_info
+    Category: com.canonical.plainbox::uncategorised
+    ... 8< -------------------------------------------------------------------------
+    interface: lo
+    link_info_kind:
+    link_type: loopback
+
+    interface: enp2s0f0
+    link_info_kind:
+    link_type: ether
+
+    interface: enp5s0
+    link_info_kind:
+    link_type: ether
+
+    interface: wlan0
+    link_info_kind:
+    link_type: ether
+
+    interface: lxdbr0
+    link_info_kind: bridge
+    link_type: ether
+
+    interface: veth993f2cd0
+    link_info_kind: veth
+    link_type: ether
+
+    interface: tun0
+    link_info_kind: tun
+    link_type: none
+
+We now add a ``requires:`` constraint to our jobs so that, if no interface
+that could possibly connected to the internet is on the machine, we can
+skip them instead of failing.
+
+.. code-block:: none
+    :emphasize-lines: 4,5
+
+    id: network_available
+    flags: simple
+    _summary: Test that the internet is reachable
+    requires:
+      network_iface_info.link_type == "ether"
+    command:
+      ping -c 1 1.1.1.1
+
+If we now run the ``network_available`` test, Checkbox will automatically pull
+also ``network_iface_info``. Note that this only happens because both are in
+the same namespace.
+
+.. code-block:: none
+
+    (checkbox_venv) > checkbox-cli run com.canonical.certification::network_available
+    ===========================[ Running Selected Jobs ]============================
+    =========[ Running job 1 / 2. Estimated time left (at least): 0:00:00 ]=========
+    ----------------[ Fetches information of all network intefaces ]----------------
+    [...]
+    =========[ Running job 2 / 2. Estimated time left (at least): 0:00:00 ]=========
+    --------------------[ Test that the internet is reachable ]---------------------
+    [...]
+    ==================================[ Results ]===================================
+     ☑ : Fetches information of all network intefaces
+     ☑ : Test that the internet is reachable
+
+Are we done then? Almost, there are a few issues with our resource job. The
+first and most relevant is that the ``resource`` constraint we have written
+seems to work, but if we analyze the output what we have written actually
+over-matches (as ``veth993f2cd0`` is also an ``ether`` device, but it is not a
+valid interface to use to connect to the internet). We can easily fix this by
+updating the expression as follows but take note of what happened.
+
+.. warning::
+    It is actually difficult to write a significant resource expressions. This
+    time we got "lucky", and we could notice the mistake on our own machine, but
+    this may not be the always the case. In general make your resource
+    expressions as restrictive as possible.
+
+.. code-block:: none
+
+    id: network_available
+    [...]
+    requires:
+      (network_iface_info.link_info_kind == "" and network_iface_info.link_type == "ether")
```

Comment:
You prefer to show them the best way to write resource expressions, instead of the following (albeit easier to read for a human)?

```suggestion
      network_iface_info.link_info_kind == ""
      network_iface_info.link_type == "ether"
```

Reply:
This is completely wrong though, this means there is at least a network_iface_info that has no link speed and one that is of type ether. the one in the tutorial means there is a network_iface_info that is of kind empty AND link type ether